collector/generic_collector.go

```diff
@@ -85,6 +82,12 @@ func NewCollector(collectorName string, configfile string) (*GenericCollector, e
 		}
 	}
 
+	// Minimum supported polling frequency is 1s.
+	minSupportedFrequency := 1 * time.Second
```

Comment:
yes, but importing HousekeepingInterval adds a cyclic dependency between manager and collector. I'll clean that up next.